test/providers/awssecrets/aws_secrets_provider_test.go

```diff
@@ -1,15 +1,158 @@
+	// mock server, String or Binary. Default is String.
+	secretStringNotBinary := true
+
+	// TODO: This should really be standalone test fixture, that maintains a map of
+	// 	request-secret-id to the latest request details and response details, or some
+	// 	other thing that is unique to a request. This will allow for arbitrarily complex
+	//	and robust testing using this mock server.
+	// NOTE: This server can and has been run standalone and consumed by the AWS CLI to
+	// confirm that it behaves as expected. Automating such a test wouldn't hurt.
+
+	// Mock server for fetching secrets. Secret value in the response is always the value
+	// "[request-secret-id]-value"
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		reqBytes, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Parse and capture the request secretInput
+		var secretInput secretsmanager.GetSecretValueInput
+		err = json.Unmarshal(reqBytes, &secretInput)
+		if err != nil {
+			t.Fatal(err)
+		}
+		secretInputs = append(secretInputs, secretInput)
+
+		// Parse request for AWS signed request metadata
+		reqMeta, err := awsinternal.NewRequestMetadata(r)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Capture original signed request header
+		originalAuthHeaders = append(originalAuthHeaders, r.Header.Get("Authorization"))
+
+		// Remove all non-signed headers
+		signedHeaders := map[string]bool{}
+		for _, name := range reqMeta.SignedHeaders {
+			signedHeaders[strings.ToLower(name)] = true
+		}
+		for name := range r.Header {
+			if ok := signedHeaders[strings.ToLower(name)]; ok {
+				continue
+			}
+
+			r.Header.Del(name)
+		}
+
+		// Re-sign request
+		_, err = v4.NewSigner(resignCreds).Sign(
+			r,
+			bytes.NewReader(reqBytes),
+			reqMeta.ServiceName,
+			reqMeta.Region,
+			reqMeta.Date,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Capture resigned request header
+		reSignedAuthHeaders = append(reSignedAuthHeaders, r.Header.Get("Authorization"))
+
+		// Craft response. The first response is secret string, then all subsequent
+		// responses are secret binary
+		var secretOutput secretsmanager.GetSecretValueOutput
+		secretValue := aws.StringValue(secretInput.SecretId) + "-value"
+		if secretStringNotBinary {
+			secretOutput.SetSecretString(secretValue)
+		} else {
+			secretOutput.SetSecretBinary([]byte(secretValue))
+		}
+
+		// Write response
+		responseBytes, err := json.Marshal(secretOutput)
+		if err != nil {
+			t.Fatal(err)
+		}
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write(responseBytes)
+	}))
+	defer server.Close()
+
+	// Setup authentication for provider
+	cleanup := setupProviderAuth()
+	defer cleanup()
+
+	// Create provider (with custom endpoint)
+	p, err := awssecrets.NewProvider(plugin_v1.ProviderOptions{Name: "aws"}, aws.Config{
+		Endpoint: aws.String(server.URL),
+	})
+	So(err, ShouldBeNil)
```

Comment:
Since the existing test has only like two assertions, and are currently style guidelines prohibit using Convey since we decided to do all testing using Go's builtin test framework way back when, I think it would be better to switch to that rather than expand the legacy code.

Reply:
Completely agree! Thanks for flagging it I'll keep in mind to step away from Convey moving forward.